docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,243 @@
+    To invoke a suffix accepting ``bytes20`` you must use one of the other literal kinds implicitly
+    convertible to ``bytes20``, e.g., a hexadecimal string literal
+    (``hex"dCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF"``).
+
+.. note::
+    Suffix functions accepting ``address payable`` are not allowed since address literals are never payable.
+
+Suffix functions may not accept or return reference types with ``storage`` or ``calldata`` locations.
+
+.. index:: function;call
+.. _calling_suffix_functions:
+
+Calling Suffix Functions
+------------------------
+
+There are two ways to call a suffix function:
+
+#. Suffix call.
+#. Function call.
+
+.. index:: ! literal suffix; suffix call syntax
+
+Suffix Call Syntax
+^^^^^^^^^^^^^^^^^^
+
+A suffix call has the same syntax as a literal with a :ref:`denomination<denominations>`:
+
+.. code-block:: solidity
+
+    42 suffix;
+    1.23 suffix;
+    "abc" suffix;
+    hex"12ff"suffix;
+    true suffix;
+
+The literal passed as input to the suffix function must be immediately followed by the name of the suffix.
+The two must be separated by whitespace unless it's a string, unicode or hexadecimal string literal,
+in which case the whitespace is optional.
+
+This call syntax supports only a single literal argument.
+Variables or expressions (even as simple as wrapping the literal in parentheses) are not allowed.
+Suffix functions defined with two parameters are also invoked with one literal - the
+:ref:`decomposition<fractional_decomposition>` of the literal into two values is performed implicitly
+by the compiler.
+
+.. note::
+    There are no negative number literals in Solidity.
+    A literal with a minus sign is an expression.
+    ``-123 suffix`` is equivalent to ``-(123 suffix)``, so ``suffix`` does not receive ``-123`` as input.
+    The argument is instead ``123``, and the negation is applied to the returned value.
+
+.. note::
+    String concatenation produces a single literal at compilation time and therefore is not treated
+    as an expression.
+    This means that e.g., ``"abc" "def" suffix`` is a valid suffix call.
+
+.. index:: ! literal suffix; function call syntax, overload
+
+Function Call Syntax
+^^^^^^^^^^^^^^^^^^^^
+
+Suffix definitions are in all respects valid free functions, and this includes the ability to call
+them directly.
+This makes it possible to call such functions with arguments which are not literals.
+
+.. code-block:: solidity
+
+    suffix(42);
+    suffix(123, 2);
+    suffix("abc");
+    suffix(hex"12ff");
+    suffix(true);
+
+Note that the fractional decomposition is not performed for this kind of call - two-argument
+suffix functions must be explicitly called with two arguments.
+
+Regardless of the call syntax used and in contrast to applying a denomination, the result of the
+call is itself not considered a literal.
+As a consequence, it cannot be used as input of another suffix call, and calculations on it are performed
```

Comment:
ah ok.

Reply:
but this is the general limitation, right? not specific for two-argument suffix functions right? maybe this paragraph should be then above the "Note that the fractional.."

Reply:
I think the current order makes more sense. That sentence refers to the example and should be close to it.